src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Sample.kt

```diff
@@ -0,0 +1,39 @@
+package uk.gov.justice.hmiprobation.casesampler.dto
+
+import uk.gov.justice.hmiprobation.casesampler.services.AllocationData
+import uk.gov.justice.hmiprobation.casesampler.utils.Size.SampleSize
+import java.time.LocalDateTime
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicInteger
+
+class RowNumberAssigner {
+    var rowCount = AtomicInteger()
+
+    fun nextId() = "${rowCount.incrementAndGet()}".padStart(3, '0')
+}
+
+data class PrimaryCaseSampleProvisional(val id: UUID, val timestamp: LocalDateTime, val results: List<StratumResult>) {
+
+    constructor(results: List<Sample>) : this(UUID.randomUUID(), LocalDateTime.now(), toStratumResults(RowNumberAssigner(), results))
+
+    companion object {
+        private fun toStratumResults(rowNumberAssigner: RowNumberAssigner, results: List<Sample>): List<StratumResult> =
```

Comment:
This is equivalent to a Java private static method?  Does that mean you can't call these functions without an instance?  So why static?  (I'm not a Scala head so companion objects are still quite new to me, please forgive me if I'm talking rubbish)

Reply:
it's because it's called by the constructor - can't call an instance method in a kotlin constructor